cassandra-migration/src/main/java/org/cognitor/cassandra/migration/Database.java

```diff
@@ -365,13 +343,12 @@ private void executeStatement(String statement, DbMigration migration) {
      * @param wasSuccessful indicates if the migration was successful or not
      */
     private void logMigration(DbMigration migration, boolean wasSuccessful) {
-        BoundStatement boundStatement = logMigrationStatement.bind(wasSuccessful, migration.getVersion(),
+        executor.execute(format(INSERT_MIGRATION, getTableName()), wasSuccessful, migration.getVersion(),
```

Comment:
logMigration query should include keyspace name. When using AWS Keyspaces, having multiple tables with the same name (even if they're in different keyspaces), sometimes makes the tool insert log entries into improper tables (or executing the query ends with an exception about insufficient access rights, depending on IAM policy configuration). Consider the following scenario:

Keyspaces: A, B, C
Tables: A.schema_migration, B.schema_migration, C.schema_migration
Query in the BoundStatement: "insert into schema_migration (applied_successful, version, script_name, script, executed_at) values(?, ?, ?, ?, ?)"

The above statement executed in the context of keyspace A will sometimes end up modifying the contents of the table B.schema_migration or C.schema_migration.

I was able to avoid this problem by modifying the query and explicitly stating the keyspace name:
"insert into %s.%s (applied_successful, version, script_name, script, executed_at) values(?, ?, ?, ?, ?)"

Not sure what is the root cause - perhaps it's because I'm using datastax driver 3.7.2 and it's not properly taking into account the context in which the query is executed? (ignoring USE keyspace?). Then again, modifying the statement here would avoid any ambiguity (related to the driver etc.)

Reply:
It is included in my last commit.